tests/rust/aligned_struct.rs

```diff
@@ -0,0 +1,84 @@
+#[repr(packed, C)]
+pub struct PackedStruct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+#[repr(align(1), C)]
+pub struct Align1Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+#[repr(align(2), C)]
+pub struct Align2Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+#[repr(align(4), C)]
+pub struct Align4Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+#[repr(align(8), C)]
+pub struct Align8Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+#[repr(align(32), C)]
+pub struct Align32Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+// Packed doesn't take an argument.
+#[repr(packed(4), C)]
+pub struct BadPacked4Struct {
+    pub arg1: usize,
+    pub arg2: *mut u8,
+}
+
+// Alignment must be an integer.
+#[repr(align("1"), C)]
```

Comment:
Please make sure that rustc compiles this. If it doesn't it's not really worth putting here IMO.

Reply:
Several of the bad alignments don't compile (including the non-power-of-2 ones). I'll drop all of those.